site/content/features/jep445.md

```diff
@@ -0,0 +1,196 @@
+* The `String[]` parameter can be omitted.
+
+That means there are twelve possible forms of `main`. Here is the shortest version of the classic “Hello, World!” program:
+
+```
+class HelloWorld {
+   void main() {
+      System.out.println("Hello, World!");
+   }
+}
+```
+
+You can launch a class that has multiple `main` methods—in fact, up to four, since you cannot overload on visibility. A `static main` method has preference over an instance method. If that leaves two candidates, the one with a `String[]` parameter has precedence.
+
+The `main` method can be declared in a superclass. This has always been true for `public static void main(String[])`. I don't see why anyone would do that, though.
+
+As with all preview features, you need to use the `--enable-preview` flag with the `java` launcher. For compiling, the flag is not needed because it was always legal to have methods of all kinds called `main`.
+
+This sandbox shows the four forms of main methods. Rename them to `main` and re-run the program to see which one is picked by the launcher.
+
+{{< sandbox version=java21 preview="true" mainclass="LauncherDemo" >}}{{< sandboxsource "LauncherDemo.java" >}}
+public class LauncherDemo {
+   static void main1(String[] args) {
+      System.out.println("static, args");
+   }
+   void main2() {
+      System.out.println("static, no args");
+   }
+   void main3(String[] args) {
+      System.out.println("instance, args");
+   }
+   void main4() {
+      System.out.println("instance, no args");
+   }
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+## Unnamed Classes
+
+A second feature of JEP 445 allows you to further shorten simple programs. When a source file contains at least one method that is declared outside the class, the compiler will place any such methods inside an *unnamed class*. The unnamed class also contains any top-level fields. Classes, interfaces, records, and enumerations that are declared in the source file are nested inside the unnamed class.
+
+Since the class is unnamed, it cannot be instantiated programmatically. The only way to instantiate it is to launch it via a `main` method. It is a compile-time error if there is no such method.
+
+Now our simplest source file becomes:
+
+{{< sandbox version=java21 preview="true" mainclass="SimplestHello" >}}{{< sandboxsource "SimplestHello.java" >}}
+void main() {
+   System.out.println("Hello, World!");
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+The class has to have some name, of course. When you place this code into a file `SimplestHello.java`, the class name is `SimplestHello`. I think one should consider that an implementation detail and launch such files with JEP 330:
+
+```
+java --enable-preview --source 21 SimplestHello.java
+```
+
+Note that you provide the source file name to the `java` launcher.
+
+You can have functions other than `main`:
+
+```
+String greeting(String greeted) {
+   return STR."Hello, \{greeted}!";
```

Comment:
Nor sure whether I would use string templates here due to the following reasons:

* Readers are just learning of new Java 21 features and might be confused be this syntax. They may wonder whether this is related to unnamed classes.
* It is not too unlikely that JEP 443 is released before JEP 430. Although it is not used in a sandbox this would still require readers to use `--enable-preview` to try these snippets.

Reply:
I thought that both JEPs are scheduled for Java 21, but I see that it is safer to disentangle the two.